sqlglot/dialects/mysql.py

```diff
@@ -463,6 +469,13 @@ class Parser(parser.Parser):
         VALUES_FOLLOWED_BY_PAREN = False
         SUPPORTS_PARTITION_SELECTION = True
 
+        def _parse_join(
+            self, skip_join_token: bool = False, parse_bracket: bool = False
+        ) -> t.Optional[exp.Join]:
+            if self._match(TokenType.STRAIGHT_JOIN, advance=False):
```

Comment:
why was this section needed? also can we get this to work with the sqlglot dialect?

Reply:
> why was this section needed?

Because we're [expecting](https://github.com/tobymao/sqlglot/blob/main/sqlglot/parser.py#L3102-L3108) the `JOIN` keyword by default, and the `STRAIGHT_JOIN` syntax doesn't have it.

> also can we get this to work with the sqlglot dialect?

Yeah, I'll refactor 👍